util/helm/client.go

```diff
@@ -355,8 +353,8 @@ func normalizeChartName(chart string) string {
 	return nc
 }
 
-func (c *nativeHelmChart) getCachedChartPath(chart string, version string) string {
-	return path.Join(c.repoPath, fmt.Sprintf("%s-%s.tgz", strings.ReplaceAll(chart, "/", "_"), version))
+func (c *nativeHelmChart) getCachedChartPath(chart string, version string) (string, error) {
+	return c.chartCachePaths.GetPath(fmt.Sprintf("%s/%s:%s", c.repoURL, strings.ReplaceAll(chart, "/", "_"), version))
```

Comment:
```suggestion
	pathKey := fmt.Sprintf("%s,%s,%s", strings.ReplaceAll(c.repoURL, ",", "\,"), strings.ReplaceAll(chart, ",", "\,"), strings.ReplaceAll(version, ",", "\,"))
	return c.chartCachePaths.GetPath(pathKey)
```

I know this is utterly painful to read, but I _think_ it 100% guarantees no collisions. :-P

Reply:
I’m not even 100% confident in this actually. Something simple that wouldn’t require error handling (like JSON would) would be base64 encoding the parts and joining on a character that can’t appear in base64.

Reply:
Implemented json encoded key as suggested in other comment